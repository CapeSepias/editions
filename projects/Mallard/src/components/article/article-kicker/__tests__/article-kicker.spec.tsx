import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { ArticleKicker } from '../normal-kicker'

describe('ArticleKicker', () => {
    it('should display a default Kicker', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ArticleKicker kicker="Sport" />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a long read style kicker', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <ArticleKicker kicker="Sport" />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
