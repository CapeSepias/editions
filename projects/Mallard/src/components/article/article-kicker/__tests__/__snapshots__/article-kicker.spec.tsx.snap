// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleKicker should display a default Kicker 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "marginBottom": 3,
        "paddingBottom": 18,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#999999",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 18,
          "lineHeight": 21,
          "marginTop": 2,
        },
        Array [
          Object {
            "color": "#121212",
          },
        ],
        Object {
          "color": "#121212",
        },
      ]
    }
  >
    Sport
  </Text>
</View>
`;

exports[`ArticleKicker should display a long read style kicker 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "marginBottom": 3,
        "paddingBottom": 18,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#999999",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 18,
          "lineHeight": 21,
          "marginTop": 2,
        },
        Array [
          Object {
            "color": "#121212",
          },
        ],
        Object {
          "color": "#121212",
        },
      ]
    }
  >
    Sport
  </Text>
</View>
`;

exports[`ArticleKicker should return null if the kicker is "Opinion" 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "marginBottom": 3,
        "paddingBottom": 18,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#999999",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "flexShrink": 0,
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 18,
          "lineHeight": 21,
          "marginTop": 2,
        },
        Array [
          Object {
            "color": "#121212",
          },
        ],
        Object {
          "color": "#121212",
        },
      ]
    }
  >
    Opinion
  </Text>
</View>
`;
